### DISCUSSION TITLE
Importing spacy v3 uses 400MB or RAM comapred to 70MB of v2

### DISCUSSION THREAD

**Discussion by HHousen** [explosion/spaCy#7107] [Help: Coding & Implementations] [answered]
Spacy v3.0.3:
```python
>>> import spacy
>>> import resource
>>> resource.getrusage(resource.RUSAGE_SELF).ru_maxrss
409804 # RAM usage in KB
```

Spacy v2.3.5:
```python
>>> import spacy
>>> import resource
>>> resource.getrusage(resource.RUSAGE_SELF).ru_maxrss
77328 # RAM usage in KB
```

Does anyone know why this is the case? Is this a bug?

**Comment by adrianeboyd** [comment, marked as answer]
On my end it looks like 57MB vs. 70MB, which doesn't seem surprising given that v3 has a few additional dependencies like pydantic.

I don't think it's a bug. Is it possible you're comparing one env with `cupy` and/or `torch` to one without? A number of libraries are imported automatically if they are available.

**Reply by HHousen** [reply]
@adrianeboyd It's the same environment. The only packages that change are `catalogue`, `srsly`, and `thinc` because spacy v2 and v3 have different requirements.
Removing `tensorflow` and `pytorch` from my environment lowers the memory usage to about 85MB. Is there a way to disable these automatic imports in v3?

**Reply by adrianeboyd** [reply]
Ah, that makes sense, since thinc v8 supports a wider range of frameworks out of the box. I only compared minimal installations of spacy in brand new venvs.

There's no way to disable the imports currently. Off the top of my head, I think it could be possible to add some global settings that control the imports, but I'm not sure whether we want the overhead of doing this.

**Reply by HHousen** [reply]
Okay, thanks for your help :smile:
